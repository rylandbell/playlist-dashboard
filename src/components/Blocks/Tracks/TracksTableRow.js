import React, { Component } from 'react';

//display '<.01' for tiny decimals, and ' ' for missing values
const formatValueDisplay = value => {
  if (typeof value !== 'number') {
    return ' ';
  } else if (value > 0 && value < 0.01) {
    return '< .01';
  } else {
    return value.toFixed(2);
  }
};

class TracksTableRow extends Component {
  constructor(props) {
    super(props);
    this.onMouseEnter = this.onMouseEnter.bind(this);
  }

  onMouseEnter() {
    this.props.handleTrackRowHover(this.props.track);
  }

  render() {
    const track = this.props.track;
    const features = this.props.features;

    return (
      <tr onMouseEnter={this.onMouseEnter}>
        <td>
          {track.name}&nbsp;
          <small>{track.artist}</small>
        </td>
        {features.map(filter =>
          <td key={filter.name} className="text-center hidden-xs">
            {formatValueDisplay(track[filter.name])}
          </td>
        )}
      </tr>
    );
  }
}

export default TracksTableRow;
