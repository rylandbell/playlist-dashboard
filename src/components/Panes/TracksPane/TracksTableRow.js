import React, { Component } from 'react';

//display '<.01' for tiny decimals, and '0.5' for missing values
const formatValueDisplay = value => {
  if (typeof value !== 'number') {
    return ' ';
  } else if (value > 0 && value < .1) {
    return '< .01';
  } else {
    return value.toFixed(2);
  }
}

class TracksTableRow extends Component {
  constructor(props) {
    super(props);
    this.onMouseEnter = this.onMouseEnter.bind(this);
  }

  onMouseEnter() {
    this.props.handleTrackRowHover(this.props.track);
  }

  render() {
    const track = this.props.track.track;
    const featuresData = this.props.featuresData;
    const allFilters = this.props.reduxState.filters;

    return (
      <tr onMouseEnter={this.onMouseEnter}> 
        <td>{track.name}&nbsp;
          <small>{track.artists[0].name}</small>
        </td>
        {allFilters.map(filter => <td key={filter.name} className="text-center hidden-xs">{formatValueDisplay(featuresData[filter.name])}</td>)}
      </tr>
    );
  }
}

export default TracksTableRow;