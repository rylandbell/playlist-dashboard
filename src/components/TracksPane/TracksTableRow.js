import React, { Component } from 'react';

class TracksTableRow extends Component {
  constructor(props) {
    super(props);
    this.onMouseEnter = this.onMouseEnter.bind(this);
  }

  onMouseEnter() {
    this.props.handleTrackRowHover(this.props.track);
  }

  render() {
    const track = this.props.track.track;
    const features = this.props.features;
    const danceability = features ? features.danceability : 'N/A';
    const instrumentalness = features ? features.instrumentalness : 'N/A';
    const valence = features ? features.valence : 'N/A';
    
    return (
      <tr className="tracks__row" onMouseEnter={this.onMouseEnter}> 
        <td>{track.name}&nbsp;
          <small>{track.artists[0].name}</small>
        </td>
        <td className="text-right">{danceability || '0'}</td>
        <td className="text-right">{instrumentalness || '0'}</td>
        <td className="text-right">{valence || '0'}</td>
      </tr>
    );
  }
}

export default TracksTableRow;