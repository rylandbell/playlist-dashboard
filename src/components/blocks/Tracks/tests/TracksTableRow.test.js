import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import TracksTableRow from "../TracksTableRow";
import { testFeaturesData, testFilteredTracks } from "./test-data";

describe("TracksTableRow component", () => {
  const props = {
  features: testFeaturesData,
  track: testFilteredTracks[0],
  handleTrackRowHover: () => {},
};

  it("renders without crashing", () => {
    const parentNode = document.createElement("tbody");
    ReactDOM.render(<TracksTableRow {...props} />, parentNode);
  });

  it("shallow render matches snapshot", () => {
    const tree = toJson(shallow(<TracksTableRow {...props} />));
    expect(tree).toMatchSnapshot();
  });
});
