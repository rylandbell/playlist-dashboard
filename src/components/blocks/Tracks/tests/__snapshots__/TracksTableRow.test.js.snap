// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TracksTableRow component shallow render matches snapshot 1`] = `
<tr
  onMouseEnter={[Function]}
>
  <td>
    Peachfuzz
    Â 
    <small>
      KMD
    </small>
  </td>
  <td
    className="text-center hidden-xs"
  >
    0.88
  </td>
  <td
    className="text-center hidden-xs"
  >
    0.84
  </td>
  <td
    className="text-center hidden-xs"
  >
    0.46
  </td>
  <td
    className="text-center hidden-xs"
  >
    0.35
  </td>
  <td
    className="text-center hidden-xs"
  >
    0.57
  </td>
  <td
    className="text-center hidden-xs"
  >
    0.00
  </td>
</tr>
`;
